src/yadism/coefficient_functions/fonll/g1_nc.py

```diff
@@ -0,0 +1,82 @@
+import numba as nb
+import numpy as np
+from ..partonic_channel import RSL
+from . import partonic_channel as pc
+from ..special.zeta import zeta2, zeta3
+from ..special.nielsen import nielsen
+from eko.constants import CA as ca
+from eko.constants import CF as cf
+from eko.constants import TR as tr
+
+# complex :math:`S_{n,m}(x)`-> nielsen(N, M, X)
+# for the light coefficient function; singlet, non singlet, gluonic 
+#LS= LNS + LPS
+# for the heavy coefficient function; singlet (only pure singlet contributes), gluonic 
+# HS= HPS
+
+@nb.njit("f8(f8,f8[:])", cache=True)
+class NonSinglet():
+    def NNLO(z, args):
+        """Massive asymptotic polarized flavor non–singlet Wilson coefficient in the MSbar scheme. 
+
+        |ref| implements :eqref:`254`, :cite:`Blümlein` (needs to be added). (coupling to the light quark)
+        """
+        L = args[0]
+        dlz=np.log(z) #H(0;x)
+        dlz2 =dlz*dlz  #H(0;x)^2
+        dlm = (-1)* np.log(1.0e0-z) #H(1;x) (relative to the definitions in other scripts, theres an extra (-1), im defining it according to 'Harmonic Polylogarithms' E. Remiddia,b and J. A. M. Vermaseren )
+        dlm2 =dlm*dlm
+
+        res= cf*tr*((-4/3)* np.power(L,2)*(1+ z) + (4/27)* (-47-218*z+36* zeta2 +36*z*zeta2)+ ((265/9)+((2*L)/3) + 2*np.power(L,2))*(np.delta(1-z)) - 4*(1 + z)*dlz2 +L *((-8/9)*(-1 + 11*z) - (8/3)*(1 + z) *dlz) - (8/9)*(5 + 14*z) * dlm - (4/3)* (1 + z)* dlm2+(-(8/9)*(13 + 28*z)-(8/3)*(1 + z)*dlm) - (8/3)*(1 + z)* nielsen(0,1,z) + 1/(1-z)*((8*np.power(L,2))/3 - (2/27)*(-359 + 144*zeta2) + 8*dlz + L*((80/9)+(16/3)*dlz) + (116/9)*dlm + (8/3)*dlm2 + dlz*((268/9) + (16/3)*dlm) + (16/3)* nielsen(0,1,z))) 
+        #remove delta later
+        return res
+    
+
+    class PureSinglet():
+        def NNLO(z, args):
+            """Massive asymptotic polarized pure singlet Wilson coefficient with in the Larin scheme. 
+
+            |ref| implements :eqref:`261`, :cite:`Blümlein` (needs to be added). (coupling to the heavy quark)
+            """
+            L = args[0]
+            dlz=np.log(z) #H(0;x)
+
+            dlm = (-1)* np.log(1.0e0-z) #H(1;x) (relative to the definitions in other scripts, theres an extra (-1), im defining it according to 'Harmonic Polylogarithms' E. Remiddia,b and J. A. M. Vermaseren )
+            dlp=np.log(1.0e0+z) #H(-1;x)
+            nH001z= None#reexpress in terms of lower weight harmonic polylogarithm
```

Comment:
If you need this functions I can provide them...

Reply:
yes, I have these two in terms of the zeta2 function and polylogs with w=1 & 2 but I think it might be best you send me since I don't have the others and they might be wrong. Thank you!

Reply:
okay sure